QualityControl/public/common/runModeHeader.js

```diff
@@ -0,0 +1,236 @@
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { RunStatus } from '../common/enums/RunStatus.enum.js';
+import { h, info } from '/js/src/index.js';
+
+/**
+ * Run mode header component showing run information and exit button
+ * @param {Model} model - root model of the application
+ * @param {FilterModel} filterModel - model that manages filter state
+ * @param {PageModel} pageModel - model that manages the state of the page
+ * @returns {vnode} - virtual node element
+ */
+export function runModeHeader(model, filterModel, pageModel) {
+  if (!model.inRunMode) {
+    return null;
+  }
+
+  return h('.run-mode-header.flex-row.items-center.p2.g2.bg-gray-lighter', {
+    id: 'run-mode-header',
+  }, [
+    renderRunModeInfo(model, filterModel),
+    h('.flex-grow'),
+    renderExitButton(filterModel, pageModel),
+  ]);
+
+  /**
+   * Renders the run mode information section
+   * @param {Model} model - root model of the application
+   * @param {FilterModel} filterModel - model that manages filter state
+   * @returns {vnode} - virtual node element
+   */
+  function renderRunModeInfo(model, filterModel) {
+    return h('.run-mode-info.flex-row.items-center.g2', {
+      id: 'run-mode-info',
+    }, [
+      renderTitle(),
+      renderRunNumber(model.runNumber),
+      renderRunStatus(model, filterModel),
+    ]);
+  }
+
+  /**
+   * Renders the run mode title
+   * @returns {vnode} - virtual node element
+   */
+  function renderTitle() {
+    return h('span.run-mode-title', {
+      id: 'run-mode-title',
+      title: 'Run Mode filters objects by a specific run number and monitors run status.' +
+      ' Other filters are disabled while active.',
+    }, 'Run Mode');
+  }
+
+  /**
+   * Renders a run detail item for run number with copy functionality
+   * @param {string} runNumber - The run number value to display
+   * @returns {vnode} - virtual node element
+   */
+  function renderRunNumber(runNumber = 'N/A') {
+    const copyToClipboard = async () => {
+      if (runNumber === 'N/A') {
+        return;
+      }
+      await navigator.clipboard.writeText(runNumber);
+      model.notification.show('Run number copied to clipboard', 'success', 1500);
+    };
+
+    return h('.run-detail.flex-row.items-center.g1', {
+      id: 'run-number',
+    }, [
+      h('span.gray-darker', {
+        id: 'run-number-label',
+      }, 'Run:'),
+      h('span.text-no-select', {
+        id: 'run-number-value',
+        title: runNumber === 'N/A' ? 'No run number available' : 'Click to copy run number',
+        onclick: copyToClipboard,
+        style: runNumber !== 'N/A' ? 'cursor: pointer; user-select: none;' : 'cursor: default;',
```

Comment:
Please use classes for styling. These `h()` elements allow you to pass classes separated by a comma as a string. There are lots of examples in the codebase.